File: guides/micronaut-kafka/analytics/java/src/main/java/example/micronaut/AnalyticsController.java
package example.micronaut;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

import java.util.List;

@Controller("/analytics")
class AnalyticsController {

    private final AnalyticsService analyticsService;

    AnalyticsController(AnalyticsService analyticsService) {
        this.analyticsService = analyticsService;
    }

    @Get // <1>
    List<BookAnalytics> listAnalytics() {
        return analyticsService.listAnalytics();
    }
}

File: guides/micronaut-kafka/analytics/kotlin/src/main/kotlin/example/micronaut/AnalyticsController.kt
package example.micronaut

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Controller("/analytics")
class AnalyticsController(private val analyticsService: AnalyticsService) {

    @Get
    fun listAnalytics(): List<BookAnalytics> = analyticsService.listAnalytics() // <1>
}
